openlibrary/plugins/upstream/adapter.py

```diff
@@ -76,22 +76,21 @@ def delegate(self, *args):
         }
 
         self.before_request()
-        try:
-            server = web.config.infobase_server
-            req = urllib.request.Request(
-                server + self.path + '?' + urllib.parse.urlencode(self.input),
-                self.data,
-                headers=headers,
-            )
-            req.get_method = lambda: web.ctx.method
-            response = urllib.request.urlopen(req)
-        except urllib.error.HTTPError as e:
-            response = e
-        self.status_code = response.code
-        self.status_msg = response.msg
-        self.output = response.read()
-
-        self.headers = dict(response.headers.items())
+        s = requests.Session()
+        server = web.config.infobase_server
+        req = requests.Request(
+            web.ctx.method,
+            server + self.path + '?' + requests.utils.requote_uri(self.input),
+            data=self.data,
+            headers=self.headers,
+        )
+        resp = s.send(req)
+        self.status_code = resp.status_code
+        # a one-liner that replaces the __str__ functionality of the HTTPResponse object replaced from urllib
```

Comment:
```suggestion
        self.status_code = resp.status_code
```
Again, I don't think we need this comment